STAR-Request/draft-sheffer-acme-star-delegation.md

```diff
@@ -0,0 +1,302 @@
+   }
+~~~
+
+The Order object that is created on the IdO:
+
+- MUST start in the "ready" state;
+- MUST contain an empty "authorizations" field;
+- MUST NOT contain the "notBefore" and "notAfter" fields.
+
+~~~
+   {
+     "status": "ready",
+     "expires": "2016-01-01T00:00:00Z",
+
+     "identifiers": [
+      {
+        "type": "dns",
+        "value": "abc.ndc.dno.example",
+        "delegated": true,
+        "cname": "abc.ndc.example"
+      }
+     ],
+
+     "authorizations": [],
+
+     "finalize": "https://acme.dno.example/acme/order/TOlocE8rfgo/finalize"
+   }
+~~~
+
+The IdO SHOULD copy any "recurrent-*" field from the NDC request into the related STAR request to the ACME CA.
+
+When the validation of the identifiers has been successfully completed and the certificate has been issued by the CA, the IdO:
+
+- MUST move its Order resource status to "valid";
+- MUST copy the "certificate" field from the STAR Order;
+
+The latter indirectly includes (via the NotBefore and NotAfter HTTP headers) the renewal timers needed by the NDC to inform its certificate reload logics.
+
+~~~
+   {
+     "status": "valid",
+     "expires": "2016-01-01T00:00:00Z",
+
+     "identifiers": [
+      {
+        "type": "dns",
+        "value": "abc.ndc.dno.example",
+        "delegated": true,
+        "cname": "abc.ndc.example"
+      }
+     ],
+
+     "authorizations": [],
+
+     "finalize": "https://acme.dno.example/acme/order/TOlocE8rfgo/finalize",
+
+     "certificate": "https://acme.ca.example/acme/order/yTyr23sSDg9"
+   }
+~~~
+
+Note that at this point in the flow, the IdO can add the CNAME records to its zone.
+
+### Order Object on the IdO-CA side
+
+The Order from the IdO to the ACME CA SHOULD strip the "cname" attribute from the twin Order from the NDC ({{sec-profile-ndc-to-ido}}), and MUST contain the necessary STAR extensions.  In addition, in order to allow the NDC to download the certificate using the unauthenticated GET, it MUST contain the recurrent-certificate-get attribute and set its value to true.
+
+### Capability Discovery
+
+In order to help a client discovering support for this profile, the directory object of an ACME server MUST contain the following attribute inside the "meta" field:
+
+- star-delegation-enabled: boolean flag indicating support for the profile specified in this memo.  An ACME server that supports this delegation profile MUST include this key, and MUST set it to true.
+
+### On Cancelation 
+
+It is worth noting that cancelation of the ACME STAR certificate is a prerogative of the IdO.  The NDC does not own the relevant account key on the ACME CA, therefore it can't issue a cancelation request for the STAR cert.  Potentially, since it holds the STAR cert private key, it could request the revocation of a single STAR certificate.  However, STAR explicitly disables the revokeCert interface.
+
+[[TBD.2: On key compromise, how does the NDC signal the IdO the need of cancelation?]]
```

Comment:
IMHO, this is explicitly out of scope (i.e., it's a phone call).

Reply:
WFM